src/main/java/com/megastore/controller/ProductController.java

```diff
@@ -29,12 +31,31 @@ public ProductController(ProductFacade productFacade) {
 
     @GetMapping("/hot")
     public ResponseEntity<List<ProductHotDto>> getHotProducts(@RequestParam(defaultValue = "4") int limit, @RequestParam(defaultValue = "1") long catId) {
-        return ResponseEntity.ok(new ArrayList<>(productFacade.findHotProducts(limit,catId)));
+        return ResponseEntity.ok(new ArrayList<>(productFacade.findHotProducts(limit, catId)));
+    }
 
+    @GetMapping("/list")
+    public ResponseEntity<Collection<ProductPLPDto>> getAllProducts(
+            @RequestParam(defaultValue = "9") int limit,
+            @RequestParam(defaultValue = "1") long subcategoryId,
+            @RequestParam(defaultValue = "UPDATED") Sorting sort,
+            @RequestParam(defaultValue = "0.00") Double priceFrom,
+            @RequestParam(defaultValue = "1000000.00") Double priceTo,
+            @RequestParam(defaultValue = "") String brand
+    ) {
+        Collection<ProductPLPDto> products;
+        if ((sort.toString()).equals("ALPHABETIC")) {
+            products = new ArrayList<>(productFacade.findAllSortedByNameASC(limit, subcategoryId, priceFrom, priceTo, brand));
+        } else if ((sort.toString()).equals("REVERSED")) {
+            products = new ArrayList<>(productFacade.findAllSortedByNameDESC(limit, subcategoryId, priceFrom, priceTo, brand));
+        } else if ((sort.toString()).equals("CHEAP")) {
+            products = new ArrayList<>(productFacade.findAllSortedByPriceASC(limit, subcategoryId, priceFrom, priceTo, brand));
+        } else if ((sort.toString()).equals("EXPENSIVE")) {
+            products = new ArrayList<>(productFacade.findAllSortedByPriceDESC(limit, subcategoryId, priceFrom, priceTo, brand));
```

Comment:
As I mentioned in the meeting, to sort a product you need to get 2 parameters:
**sortBy**: "price" or "alphabetical"
**orderBy**: "ASC" or "DESC"
The first parameter indicates what type to sort by.
The second parameter tells you in which direction you will sort.